mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -717,6 +719,49 @@ class SparseVector(
       new SparseVector(size, ii, vv)
     }
   }
+
+  override def argmax: Int = {
+    if (size == 0) {
+      -1
+    } else {
+
+      //grab first active index and value by default
+      var maxIdx = indices(0)
+      var maxValue = values(0)
+
+      foreachActive { (i, v) =>
+        if (v > maxValue) {
+          maxIdx = i
+          maxValue = v
+        }
+      }
+
+      // look for inactive values incase all active node values are negative
+      if(size != values.size && maxValue <= 0){
```

Comment:
> Found another corner case with if we have a 0 value defined in the active set of values at the very end of the vector. 

Wouldn't this logic deal with all such cases?

    if(size != values.size && maxValue <= 0) {
            // calculate first inactive value
            if (maxValue == 0) {
                if (firstInactiveValue > maxIdx) maxIdx else firstInactiveValue
            }
            else {
                maxIdx
            }
    }